### PR TITLE
docs(development): clarify logging principles

### PR DESCRIPTION
**Pull request by fgreinacher**
## Changes

Tries to make the logging principles a bit clearer.

## Context

See https://github.com/renovatebot/renovate/pull/28071#discussion_r1541101705

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

:hammer_and_pick: with :heart: by [Siemens](https://opensource.siemens.com/)

### CODE REVIEW

File: docs/development/best-practices.md [hunk 1]
```diff
@@ -157,11 +157,11 @@ Avoid refactoring the code and tests at the same time, this can mask regression
 For `WARN`, `ERROR` and `FATAL` log messages use logger metadata.
 Also use logger metadata the result is a complex metadata object needing a multiple-line pretty stringification.
 
-For `INFO` log messages inline the metadata into the log message.
+For `INFO`, `DEBUG` and `TRACE` log messages inline the metadata into the log message.
 Also inline the metadata if the metadata object is complex.
```
Comment (fgreinacher) [inline docs/development/best-practices.md:161]: I do not understand this sentence. It sounds contradictory to `use logger metadata the result is a complex metadata object` for me.

File: docs/development/best-practices.md [hunk 2]
```diff
@@ -157,11 +157,11 @@ Avoid refactoring the code and tests at the same time, this can mask regression
...
 
 `WARN`, `ERROR` and `FATAL` messages are often used in metrics or error catching services.
-These log messages should have a consistent `msg` component, so they can be automatically grouped or associated.
```
Comment (fgreinacher) [inline docs/development/best-practices.md:164]: I think static is more important than consistent here.

File: docs/development/best-practices.md [hunk 3]
```diff
@@ -157,11 +157,11 @@ Avoid refactoring the code and tests at the same time, this can mask regression
...
+These log messages should have a static `msg` component, so they can be automatically grouped or associated.
 Metadata that's separate from its message is hard for humans to read.
 Try to combine the metadata into the message, unless it's too complex to do so.
```
Comment (fgreinacher) [inline docs/development/best-practices.md:166]: This is also somewhat contradictory to what's stated above if it's meant to apply for all log levels.

File: docs/development/best-practices.md (hunk 1 above)
Comment (rarkins) [reply docs/development/best-practices.md:161]: Yes I think it should say something like "It is permissible to not inline metadata if it's complex, but also think first if so much data is necessary to be logged".

File: docs/development/best-practices.md [hunk 4]
```diff
@@ -157,11 +157,11 @@ Avoid refactoring the code and tests at the same time, this can mask regression
 For `WARN`, `ERROR` and `FATAL` log messages use logger metadata.
 Also use logger metadata the result is a complex metadata object needing a multiple-line pretty stringification.
 
-For `INFO` log messages inline the metadata into the log message.
+For `INFO`, `DEBUG` and `TRACE` log messages inline the metadata into the log message.
```
Comment (rarkins) [inline docs/development/best-practices.md:160]:
```suggestion
For `INFO` and `DEBUG` log messages inline the metadata into the log message where feasible.
```

It's rare to look at TRACE messages so we don't need to care how they are structured

File: docs/development/best-practices.md (hunk 3 above)
Comment (rarkins) [reply docs/development/best-practices.md:166]: Yes, please remove this sentence too

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**